src/Autodesk.Forge.Core/ForgeHandler.cs

```diff
@@ -88,33 +95,34 @@ protected virtual IAsyncPolicy<HttpResponseMessage> GetTokenRefreshPolicy()
                 );
         }
 
-        protected virtual IAsyncPolicy<HttpResponseMessage> GetResiliencyPolicies(int timeoutSeconds = ForgeGatewayTimeoutSeconds)
+        protected virtual IAsyncPolicy<HttpResponseMessage> GetResiliencyPolicies(TimeSpan timeoutValue)
         {
             // Retry when HttpRequestException is thrown (low level network error) or 
             // the server returns an error code that we think is transient
-            var errors = Policy
-                .Handle<HttpRequestException>()
-                .OrResult<HttpResponseMessage>(response =>
-                {
-                    int[] retriable = {
+            //
+            int[] retriable = {
                         (int)HttpStatusCode.RequestTimeout, // 408
                         429, //too many requests
                         (int)HttpStatusCode.BadGateway, // 502
                         (int)HttpStatusCode.ServiceUnavailable, // 503
                         (int)HttpStatusCode.GatewayTimeout // 504
                         };
+            var (retryBaseDelay, retryMultiplier) = GetRetryParameters();
+            var retry = Policy
+                .Handle<HttpRequestException>()
+                .Or<Polly.Timeout.TimeoutRejectedException>()// thrown by Polly's TimeoutPolicy if the inner call times out
```

Comment:
from the log, I also see the exception of `System.Threading.Tasks.TaskCanceledException`, not sure what is the order of these two exceptions and which one should be caught..

Reply:
`TaskCancelledExecption` will be thrown by the cancelled async operation (i.e. SendAsync) in our case. It will be caught by our Timeout policy (if the cancel was caused by our timeout policy) and rethrown as `TimeoutRejectedException`.